nilearn/image/tests/test_image.py

```diff
@@ -274,3 +275,63 @@ def test_swap_img_hemispheres():
 
 def test_concat_imgs():
     assert_true(concat_imgs is niimg_conversions.concat_niimgs)
+
+
+def test_index_img():
+    img_3d = nibabel.Nifti1Image(np.ones((3, 4, 5)), np.eye(4))
+    assert_raises_regexp(TypeError, '4D Nifti image',
+                         image.index_img, img_3d, 0)
+
+    affine = np.array([[1., 2., 3., 4.],
+                       [5., 6., 7., 8.],
+                       [9., 10., 11., 12.],
+                       [0., 0., 0., 1.]])
+    img_4d, _ = testing.generate_fake_fmri(affine=affine)
+
+    fourth_dim_size = niimg_conversions._get_shape(img_4d)[3]
+    tested_indices = (range(fourth_dim_size) +
+                      [slice(2, 8, 2), [1, 2, 3, 2],
+                       (np.arange(fourth_dim_size) % 3) == 1])
+    for i in tested_indices:
+        this_img = image.index_img(img_4d, i)
+        expected_data_3d = img_4d.get_data()[..., i]
+        assert_array_equal(this_img.get_data(),
+                           expected_data_3d)
+        assert_array_equal(this_img.get_affine(),
+                           img_4d.get_affine())
+
+    for i in [fourth_dim_size, - fourth_dim_size - 1,
```

Comment:
Done a few lines above. `index=[]` is actually a valid index, looks like you were expecting an out of bounds error from the positioning of your comment.